public/src/pages/views/EnrollmentView.tsx

```diff
@@ -13,21 +13,21 @@ export class EnrollmentView extends React.Component<IEnrollmentViewProps, {}> {
         return <DynamicTable
             data={this.props.courses}
             header={["Course code", "Course Name", "Action"]}
-            selector={(course: IAssignmentLink) => this.createEnrollmentRow(course)}>
+            selector={(course: IStudentLabsForCourse) => this.createEnrollmentRow(course)}>
         </DynamicTable>;
 
     }
 
-    public createEnrollmentRow(course: IAssignmentLink):
-        (string | JSX.Element)[] {
-        const base: (string | JSX.Element)[] = [course.course.getCode(), course.course.getName()];
-        if (course.link) {
-            if (course.link.getStatus() === Enrollment.UserStatus.STUDENT
-                 || course.link.getStatus() === Enrollment.UserStatus.TEACHER) {
+    public createEnrollmentRow(course: IStudentLabsForCourse):
+        Array<string | JSX.Element> {
+        const base: Array<string | JSX.Element> = [course.course.getCode(), course.course.getName()];
+        if (course.enrollment) {
```

Comment:
These if-statements can be simplified into methods on `IStudentLabsForCourse`, e.g. `course.isStudent()` and `course.isTeacher()`... although I'm not too happy with the `course` naming here...